scripts/test_cases/msbuild_performance_test_linux.py

```diff
@@ -0,0 +1,171 @@
+    subprocess.call("ls", shell=True)
+
+
+def create_extract_destinations():
+    """ Create the extract destination directories if they do not exist"""
+    check_directory('BEFORE CREATE EXTRACT DESTINATION')
+    if not os.path.exists(EXTRACT_PATH):
+        os.mkdir(EXTRACT_PATH)
+    os.chdir(EXTRACT_PATH)
+    if not os.path.exists("base"):
+        os.mkdir("base")
+    if not os.path.exists("daily"):
+        os.mkdir("daily")
+    check_directory('AFTER CREATE EXTRACT DESTINATION')
+
+
+def download_file(url, filename):
+    """ Download file from url and save it to filename"""
+    check_directory('BEFORE DOWNLOAD FILE')
+    with urllib.request.urlopen(url) as response, open(filename, 'wb') as out_file:
+        data = response.read()
+        out_file.write(data)
+    check_directory('AFTER DOWNLOAD FILE')
+
+
+def download_and_extract_dotnet_sdk(version_url, is_base):
+    """ Download and extract the dotnet sdk"""
+    path = "base" if is_base else "daily"
+    tar_gz_file = "dotnet-sdk.tar.gz"
+    download_file(version_url, tar_gz_file)
+
+    # Extract the tar.gz file
+    check_directory('BEFORE EXTRACT FILE')
+    extract_command = f"tar -xzf {tar_gz_file} -C {path}"
+    subprocess.call(extract_command, shell=True)
+
+
+def measure_execution_time(command):
+    """summary for measure_execution_time
+
+    Args:
+        command ([String]): [command to be executed]
+
+    Returns:
+        [Number]: [ellapsed time in seconds]
+    """
+
+    check_directory('BEFORE MEASURE EXECUTION TIME')
+    subprocess.call(f"cd {TEST_REPO_NAME}", shell=True)
+    check_directory('AFTER CHECKINT OUT TO REPO')
+
+    # Record start time
+    start_time = time.time()
+
+    # Run the command
+    subprocess.call(command, shell=True)
+
+    # Calculate elapsed time
+    end_time = time.time()
+    elapsed_time = end_time - start_time
+
+    return elapsed_time
+
+
+def clone_repository(repo_url, repo_path):
+    """_summary_
+
+    Args:
+        repo_url (String): url of the repository to be cloned
+        repo_path (String): path containing test code
+    """
+    check_directory('BEFORE CLONE REPO')
+    # Clone the repository containing the solution
+    subprocess.call(f"git clone {repo_url}", shell=True)
+    subprocess.call(f"cd {TEST_REPO_NAME}", shell=True)
+    subprocess.call(f"cd {repo_path}", shell=True)
+    subprocess.call(f"cd {TEST_SOLUTION_DIR}", shell=True)
+    check_directory('AFTER CLONE REPO')
+    subprocess.call("ls", shell=False)
+
+
+def delete_clone(repo_url):
+    """_summary_
+
+    Args:
+        repo_path (String): path containing test code
+    """
+    # extract the repository name from the url
+    repo_name = repo_url.split("/")[-1].replace(".git", "")
+    subprocess.call(f"rm -rf {repo_name}", shell=True)
+
+
+def main():
+    """_summary_
+        main()
+
+    """
+    # create a working directory for the test experiment
+    if not os.path.exists(WORKING_DIR):
```

Comment:
Is it necessary to create another "working dir"? can't we just use the current directory as a working directory?